python/dolfinx/wrappers/la.cpp

```diff
@@ -285,6 +285,11 @@ void la(nb::module_& m)
           nb::rv_policy::reference_internal);
 
   // Declare objects that are templated over type
+  declare_objects<std::int8_t>(m, "int8");
```

Comment:
Do we really want all these types at this stage? There is a compile time cost.

Does the MPI scatter support each the types?

Reply:
Good point. Looks like the types I've added in this PR are supported: https://github.com/FEniCS/dolfinx/blob/main/cpp/dolfinx/common/MPI.h#L273-L304, which are used in `Scatterer`.

Perhaps I went overboard with useful types and I agree regarding compile times. Could we go with `int32`, `int64` and something to efficiently represent a true/false? I can discuss with @IgorBaratta whether that be `int8` or `bool`?

Reply:
I would suggest `int8`. Seems that NumPy uses 1 byte for `bool`, so no space saving with `bool` over `int8`.